backend/src/gh.rs

```diff
@@ -445,4 +583,117 @@ impl GitHubClient {
+    /// - If the GET request to fetch repository details fails (e.g., network issues or API errors).
+    /// - If the response does not contain a valid `default_branch` field.
+    ///
+    /// # Returns
+    /// - `Ok(String)` containing the default branch name if successful.
+    /// - `Err(anyhow::Error)` if any step in the process fails.
+    pub async fn get_default_branch(&self) -> Result<String> {
+        // Extract repository name from `repo_url`
+        let repo_name = self.get_repo_name()?;
+    
+        // Make the GET request to fetch repository details
+        let response = self
+            .client
+            .get(format!("{}/repos/{}", GITHUB_API_URL, repo_name))
+            .bearer_auth(&self.token)
+            .header("User-Agent", "Hyde")
+            .send()
+            .await?;
+    
+        // Check response status
+        if !response.status().is_success() {
+            let status = response.status();
+            let response_text = response.text().await?;
+            bail!("Failed to fetch repository details: {}, Response: {}", status, response_text);
+        }
+    
+        // Deserialize the response to get the repository details
+        let repo_details: serde_json::Value = response.json().await?;
+    
+        // Retrieve the default branch from the response
+        let default_branch = repo_details["default_branch"]
+            .as_str()
+            .map(ToString::to_string)
+            .context("'default_branch' field missing in the response")?;
+    
+        Ok(default_branch)
+    }        
+
+    /// Fetches issues from the GitHub repository.
+    ///
+    /// This function retrieves issues from the specified repository using the GitHub API.
+    /// You can filter issues based on their state and associated labels.
+    ///
+    /// # Parameters:
+    /// - `state`: A string slice representing the state of the issues to fetch (e.g., "open", "closed", "all").
+    ///            Defaults to "open".
+    /// - `labels`: A comma-separated string slice representing labels to filter issues by. Defaults to `None`.
+    ///
+    /// # Returns:
+    /// A `Result<Vec<Value>>`:
+    /// - `Ok(issues)`: A vector of JSON values representing the issues fetched from the repository.
+    /// - `Err(e)`: An error message if the request fails or the response cannot be parsed.
+    ///
+    /// # Errors:
+    /// This function may return an error if:
+    /// - The `repo_url` is not in the expected format and cannot be parsed to derive the repository name.
+    /// - The request to fetch issues fails due to authentication issues, invalid input, or network problems.
+    /// - The GitHub API response cannot be parsed as a JSON array.
+    pub async fn get_issues(&self, state: Option<&str>, labels: Option<&str>) -> Result<Vec<Value>> {
+        let repo_name = self.get_repo_name().map_err(|e| {
+            error!("Failed to get repository name: {:?}", e);
+            e
+        })?;
```

Comment:
This is inconsistent with everywhere else, where you don't attach any additional context. I'd personally say that here you don't need to include any additional context, because the error message will point to `get_repo_name` failing, which should hopefully be apparent enough.